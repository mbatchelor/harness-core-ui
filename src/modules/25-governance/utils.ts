import type { GovernanceMetadata } from 'services/cd-ng'

interface DoesGovernanceMetaDataHasErrorOrWarningRtn {
  governanceMetaDataHasError: boolean
  governanceMetaDataHasWarning: boolean
}
export const doesGovernanceHasErrorOrWarning = (
  governanceMetadata: GovernanceMetadata | undefined
): DoesGovernanceMetaDataHasErrorOrWarningRtn => {
  let governanceMetaDataHasWarning = false
  let governanceMetaDataHasError = false
  if (governanceMetadata && governanceMetadata?.status === 'error') {
    governanceMetaDataHasError = true
  }
  if (governanceMetadata && governanceMetadata?.status === 'warning') {
    governanceMetaDataHasWarning = true
  }
  return { governanceMetaDataHasError, governanceMetaDataHasWarning }
}
