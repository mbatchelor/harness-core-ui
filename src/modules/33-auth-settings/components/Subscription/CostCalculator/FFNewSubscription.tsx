/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useState } from 'react'
import { Slider } from '@blueprintjs/core'
import { DropDown, Layout, Text, TextInput, SelectOption, PageError, Container } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import type { UsageAndLimitReturn } from '@common/hooks/useGetUsageAndLimit'
import { Editions } from '@common/constants/SubscriptionTypes'
import { ContainerSpinner } from '@common/components/ContainerSpinner/ContainerSpinner'
import css from './CostCalculator.module.scss'

const SliderWrapper = ({
  min,
  max,
  stepSize,
  labelStepSize,
  value,
  onChange
}: {
  min: number
  max: number
  stepSize: number
  labelStepSize: number
  value: number
  onChange: (value: number) => void
}): React.ReactElement => {
  return (
    <Slider
      className={css.slider}
      min={min}
      max={max}
      stepSize={stepSize}
      value={value}
      onChange={onChange}
      labelStepSize={labelStepSize}
    />
  )
}

const SliderBar = ({
  title,
  min,
  max,
  stepSize,
  labelStepSize,
  list,
  value,
  setValue,
  unit
}: {
  title: React.ReactElement
  min: number
  max: number
  stepSize: number
  labelStepSize: number
  list?: number[]
  value: number
  setValue: (value: number) => void
  unit?: string
}): React.ReactElement => {
  const dropdownList = React.useMemo(() => {
    return (
      list?.reduce((accumulator: SelectOption[], current) => {
        accumulator.push({
          label: `${current.toString()}${unit}`,
          value: current + ''
        } as SelectOption)
        return accumulator
      }, []) || []
    )
  }, [list, unit])

  return (
    <Layout.Vertical width={'50%'} padding={{ right: 'xlarge' }}>
      {title}
      <Layout.Horizontal spacing={'large'}>
        <SliderWrapper
          min={min}
          max={max}
          stepSize={stepSize}
          value={value}
          onChange={newValue => {
            setValue(newValue)
          }}
          labelStepSize={labelStepSize}
        />
        {list ? (
          <DropDown
            buttonTestId="slider-dropdown"
            filterable={false}
            width={70}
            items={dropdownList}
            value={value.toString()}
            onChange={selected => setValue(Number(selected.value))}
            placeholder={`  ${unit}`}
            className={css.dropdown}
          />
        ) : (
          <TextInput
            onChange={(e: React.ChangeEvent<HTMLInputElement>) => setValue(Number(e.target.value))}
            value={value.toString()}
            className={css.textInput}
          />
        )}
      </Layout.Horizontal>
    </Layout.Vertical>
  )
}

export const FFNewSubscription = ({
  plan,
  usageAndLimitInfo
}: {
  plan: Editions
  usageAndLimitInfo: UsageAndLimitReturn
}): React.ReactElement => {
  const { getString } = useStrings()

  const { usageData } = usageAndLimitInfo
  const { usage } = usageData
  const activeLicenses = usage?.ff?.activeFeatureFlagUsers?.count || 0
  const activeMAUs = usage?.ff?.activeClientMAUs?.count || 0

  const [licenses, setLicenses] = useState<number>(activeLicenses)
  const [maus, setMaus] = useState<number>(activeMAUs)

  const [licenseRange, setLicensesRange] = useState<{
    min: number
    max: number
    stepSize: number
    labelStepSize: number
  }>({ min: 0, max: 0, stepSize: 1, labelStepSize: 1 })
  const [mausRange, setMausRange] = useState<{
    min: number
    max: number
    stepSize: number
    labelStepSize: number
    list: number[]
    unit: string
  }>({
    min: 0,
    max: 0,
    stepSize: 1,
    labelStepSize: 1,
    list: [],
    unit: ''
  })

  const { loadingUsage, usageErrorMsg, refetchUsage } = usageData

  useEffect(() => {
    // TODO: get tier from prices api call
    if (plan === Editions.TEAM) {
      setLicensesRange({
        min: 0,
        max: 50,
        stepSize: 1,
        labelStepSize: 10
      })
      setMausRange({
        min: 0,
        max: 500,
        stepSize: 100,
        labelStepSize: 100,
        list: [100, 200, 300, 400, 500],
        unit: 'K'
      })
      // reset licenses and maus
      setLicenses(Math.max(activeLicenses, 0))
      setMaus(Math.max(activeMAUs, 0))
    } else {
      setLicensesRange({
        min: 25,
        max: 100,
        stepSize: 1,
        labelStepSize: 25
      })
      setMausRange({
        min: 0,
        max: 25,
        stepSize: 5,
        labelStepSize: 5,
        list: [0, 5, 10, 15, 20, 25],
        unit: 'M'
      })
      // reset licenses and maus
      setLicenses(Math.max(activeLicenses, 25))
      setMaus(Math.max(activeMAUs, 0))
    }
  }, [plan, activeLicenses, activeMAUs])

  if (loadingUsage) {
    return <ContainerSpinner />
  }

  if (usageErrorMsg) {
    return (
      <Container width={300}>
        <PageError message={usageErrorMsg} onClick={refetchUsage} />
      </Container>
    )
  }

  const mausTitle = (
    <Layout.Horizontal flex={{ alignItems: 'baseline', justifyContent: 'start' }}>
      <Text font={{ weight: 'bold' }}>{getString('authSettings.costCalculator.maus')}</Text>
      <Text font={{ size: 'small' }}>
        , {mausRange.stepSize}
        {mausRange.unit} {getString('authSettings.costCalculator.increments')}
      </Text>
    </Layout.Horizontal>
  )

  return (
    <Layout.Horizontal>
      <SliderBar
        title={<Text font={{ weight: 'bold' }}>{getString('authSettings.costCalculator.developerLicenses')}</Text>}
        min={licenseRange.min}
        max={licenseRange.max}
        stepSize={licenseRange.stepSize}
        labelStepSize={licenseRange.labelStepSize}
        value={licenses}
        setValue={setLicenses}
      />
      <SliderBar
        title={mausTitle}
        min={mausRange.min}
        max={mausRange.max}
        stepSize={mausRange.stepSize}
        labelStepSize={mausRange.labelStepSize}
        list={mausRange.list}
        value={maus}
        setValue={setMaus}
        unit={mausRange.unit}
      />
    </Layout.Horizontal>
  )
}
