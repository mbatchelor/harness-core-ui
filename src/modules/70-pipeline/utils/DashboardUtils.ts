/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { TooltipFormatterContextObject } from 'highcharts'

interface PointStats {
  deployments?: {
    failure?: number
    success?: number
    total?: number
  }
  time?: number
}

/* istanbul ignore next */
export const renderTooltipContent = ({
  time,
  failureRate,
  count,
  successCount,
  failureCount
}: {
  time: string | number
  failureRate: string | number
  count?: number
  successCount?: number
  failureCount?: number
}) => {
  return `<div style="padding: 16px; color: white; width: 282px; height: 128px;">
        <div style="display: flex; justify-content: space-between; border-bottom: 0.5px solid rgba(243, 243, 250); padding-bottom: 7px; margin-bottom: 15px;">
          <div style="font-weight: normal; font-size: 12px; line-height: 18px; opacity: 0.8;">${time}</div>
          <div>
            <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Deployments: </span>
            <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${count}</span>
          </div>
        </div>
        <div style="display: flex; justify-content: space-between;">
          <div>
            <p style="font-weight: 500; font-size: 10px; line-height: 14px; letter-spacing: 0.2px; color: #D9DAE6; margin-bottom: 0px;">Failure Rate</p>
            <p style="font-weight: 600; font-size: 28px; line-height: 38px; color: #FBE6E4;">${failureRate}</p>
          </div>
          <div style="margin-right: 8px;">
            <div style="display: flex; align-items: center; margin-bottom: 6px;">
              <div style="height: 6px; width: 12px; background-color: #5FB34E; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px; opacity: 0.8;">Success </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 16px; letter-spacing: 0.2px;">${successCount}</span>
            </div>
            <div style="display: flex; align-items: center;">
              <div style="height: 6px; width: 12px; background-color: #EE5F54; border-radius: 16px; display: inline-block; margin-right: 8px"></div>
              <span style="white-space: pre; font-weight: bold; font-size: 12px; line-height: 18px; opacity: 0.8;">Failed </span>
              <span style="font-weight: bold; font-size: 12px; line-height: 18px;">${failureCount}</span>
            </div>
          </div>
        </div>
      </div>`
}

/* istanbul ignore next */
export const getTooltip = (currPoint: TooltipFormatterContextObject): string => {
  const custom = currPoint?.series?.userOptions?.custom
  const point: PointStats = custom?.[currPoint.key]
  const time =
    point && point?.time
      ? new Date(point?.time).toLocaleDateString('en-US', { day: 'numeric', month: 'long' })
      : currPoint.x
  let failureRate: string | number = 'Infinity'
  if (point?.deployments?.failure && point.deployments?.total) {
    failureRate = ((point.deployments.failure / point.deployments.total) * 100).toFixed(1) + '%'
  }
  if (point?.deployments?.failure === 0) {
    failureRate = '0'
  }
  return renderTooltipContent({
    time,
    failureRate,
    count: point?.deployments?.total,
    successCount: point?.deployments?.success,
    failureCount: point?.deployments?.failure
  })
}
