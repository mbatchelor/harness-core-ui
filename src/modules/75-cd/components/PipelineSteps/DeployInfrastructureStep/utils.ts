/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormikProps } from 'formik'
import { isEmpty } from 'lodash-es'

import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { EnvironmentResponseDTO, PipelineInfrastructure } from 'services/cd-ng'

import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'

export interface DeployInfrastructureProps {
  initialValues: PipelineInfrastructure
  onUpdate?: (data: PipelineInfrastructure) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: PipelineInfrastructure
    path?: string
    readonly?: boolean
  }
}

export interface DeployInfrastructureState {
  isEdit: boolean
  isEnvironment: boolean
  formik?: FormikProps<PipelineInfrastructure>
  data?: EnvironmentResponseDTO
}

export function isEditEnvironment(data: PipelineInfrastructure): boolean {
  if (getMultiTypeFromValue(data.environmentRef) !== MultiTypeInputType.RUNTIME && !isEmpty(data.environmentRef)) {
    return true
  } else if (data.environment && !isEmpty(data.environment.identifier)) {
    return true
  }
  return false
}

export function isEditInfrastructure(data: any): boolean {
  if (
    getMultiTypeFromValue(data.infrastructureRef) !== MultiTypeInputType.RUNTIME &&
    !isEmpty(data.infrastructureRef)
  ) {
    return true
  } else if (data.infrastructure && !isEmpty(data.infrastructure.identifier)) {
    return true
  }
  return false
}
