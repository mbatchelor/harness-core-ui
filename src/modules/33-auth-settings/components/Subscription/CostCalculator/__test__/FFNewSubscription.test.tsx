/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { Editions } from '@common/constants/SubscriptionTypes'
import { FFNewSubscription } from '../FFNewSubscription'

describe('FFNewSubscription', () => {
  test('update licenses', async () => {
    const usageAndLimitInfo = {
      limitData: {
        limit: {
          ff: {
            totalFeatureFlagUnits: 250,
            totalClientMAUs: 100000
          }
        }
      },
      usageData: {
        usage: {
          ff: {
            activeFeatureFlagUsers: {
              count: 20
            },
            activeClientMAUs: {
              count: 10000
            }
          }
        }
      }
    }

    const { container, getByRole } = render(
      <TestWrapper>
        <FFNewSubscription plan={Editions.TEAM} usageAndLimitInfo={usageAndLimitInfo} />
      </TestWrapper>
    )
    await fireEvent.change(getByRole('textbox'), { target: { value: '35' } })
    expect(container).toMatchSnapshot()
  })

  test('loading', () => {
    const usageAndLimitInfo = {
      limitData: {
        limit: {
          ff: {
            totalFeatureFlagUnits: 250,
            totalClientMAUs: 100000
          }
        }
      },
      usageData: {
        loadingUsage: true
      }
    }

    const { container } = render(
      <TestWrapper>
        <FFNewSubscription plan={Editions.TEAM} usageAndLimitInfo={usageAndLimitInfo} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('error', () => {
    const refetchMock = jest.fn()
    const usageAndLimitInfo = {
      limitData: {
        limit: {
          ff: {
            totalFeatureFlagUnits: 250,
            totalClientMAUs: 100000
          }
        }
      },
      usageData: {
        usageErrorMsg: 'error',
        refetchUsage: refetchMock
      }
    }

    const { container, getByText } = render(
      <TestWrapper>
        <FFNewSubscription plan={Editions.TEAM} usageAndLimitInfo={usageAndLimitInfo} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
    fireEvent.click(getByText('Retry'))
    expect(refetchMock).toHaveBeenCalledTimes(1)
  })

  test('dropdown', () => {
    const usageAndLimitInfo = {
      limitData: {
        limit: {
          ff: {
            totalFeatureFlagUnits: 250,
            totalClientMAUs: 100000
          }
        }
      },
      usageData: {
        usage: {
          ff: {
            activeFeatureFlagUsers: {
              count: 20
            },
            activeClientMAUs: {
              count: 10000
            }
          }
        }
      }
    }

    const { container } = render(
      <TestWrapper>
        <FFNewSubscription plan={Editions.TEAM} usageAndLimitInfo={usageAndLimitInfo} />
      </TestWrapper>
    )

    const filterDropdown = container.querySelector('[data-testid="slider-dropdown"]')
    if (filterDropdown) {
      fireEvent.click(filterDropdown)
    }

    const listItem = document.body.getElementsByClassName('DropDown--menuItem')[0]
    fireEvent.click(listItem)
    expect(listItem).toHaveTextContent('100')
    expect(container).toMatchSnapshot()
  })
})
