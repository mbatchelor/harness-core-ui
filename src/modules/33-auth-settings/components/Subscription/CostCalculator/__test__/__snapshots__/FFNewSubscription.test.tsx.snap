// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FFNewSubscription dropdown 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main"
  >
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-right-xlarge"
      style="width: 50%;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-bold"
      >
        authSettings.costCalculator.developerLicenses
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
      >
        <div
          class="bp3-slider slider"
        >
          <div
            class="bp3-slider-track"
          >
            <div
              class="bp3-slider-progress"
              style="left: 0.00%; right: 100.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress bp3-intent-primary"
              style="left: 0.00%; right: 60.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress"
              style="left: 40.00%; right: 0.00%; top: 0px;"
            />
          </div>
          <div
            class="bp3-slider-axis"
          >
            <div
              class="bp3-slider-label"
              style="left: 0.00%;"
            >
              0
            </div>
            <div
              class="bp3-slider-label"
              style="left: 20.00%;"
            >
              10
            </div>
            <div
              class="bp3-slider-label"
              style="left: 40.00%;"
            >
              20
            </div>
            <div
              class="bp3-slider-label"
              style="left: 60.00%;"
            >
              30
            </div>
            <div
              class="bp3-slider-label"
              style="left: 80.00%;"
            >
              40
            </div>
            <div
              class="bp3-slider-label"
              style="left: 100.00%;"
            >
              50
            </div>
          </div>
          <span
            class="bp3-slider-handle"
            style="left: calc(40.00% - 0px);"
            tabindex="0"
          >
            <span
              class="bp3-slider-label"
            >
              20
            </span>
          </span>
        </div>
        <div
          class="TextInput--main"
        >
          <div
            class="bp3-input-group textInput"
          >
            <input
              class="bp3-input"
              type="text"
              value="20"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-right-xlarge"
      style="width: 50%;"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline StyledProps--flex-justifyContent-start"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-weight-bold"
        >
          authSettings.costCalculator.maus
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-small"
        >
          , 
          100
          K
           
          authSettings.costCalculator.increments
        </p>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
      >
        <div
          class="bp3-slider slider"
        >
          <div
            class="bp3-slider-track"
          >
            <div
              class="bp3-slider-progress"
              style="left: 0.00%; right: 100.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress bp3-intent-primary"
              style="left: 0.00%; right: 80.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress"
              style="left: 20.00%; right: 0.00%; top: 0px;"
            />
          </div>
          <div
            class="bp3-slider-axis"
          >
            <div
              class="bp3-slider-label"
              style="left: 0.00%;"
            >
              0
            </div>
            <div
              class="bp3-slider-label"
              style="left: 20.00%;"
            >
              100
            </div>
            <div
              class="bp3-slider-label"
              style="left: 40.00%;"
            >
              200
            </div>
            <div
              class="bp3-slider-label"
              style="left: 60.00%;"
            >
              300
            </div>
            <div
              class="bp3-slider-label"
              style="left: 80.00%;"
            >
              400
            </div>
            <div
              class="bp3-slider-label"
              style="left: 100.00%;"
            >
              500
            </div>
          </div>
          <span
            class="bp3-slider-handle"
            style="left: calc(20.00% - 0px);"
            tabindex="0"
          >
            <span
              class="bp3-slider-label"
            >
              100
            </span>
          </span>
        </div>
        <div
          class="bp3-popover-wrapper DropDown--main dropdown"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="slider-dropdown"
                style="width: 70px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                    100K
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="bp3-overlay bp3-overlay-inline"
          >
            <div
              class="bp3-transition-container bp3-popover-exit bp3-popover-exit-active"
              style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
            >
              <div
                class="bp3-popover bp3-minimal bp3-select-popover DropDown--popover"
              >
                <div
                  class="bp3-popover-content"
                >
                  <div>
                    <ul
                      class="bp3-menu"
                    >
                      <li
                        class="DropDown--menuItem DropDown--active"
                      >
                        100K
                      </li>
                      <li
                        class="DropDown--menuItem"
                      >
                        200K
                      </li>
                      <li
                        class="DropDown--menuItem"
                      >
                        300K
                      </li>
                      <li
                        class="DropDown--menuItem"
                      >
                        400K
                      </li>
                      <li
                        class="DropDown--menuItem"
                      >
                        500K
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FFNewSubscription error 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    style="width: 300px;"
  >
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
      style="width: 100%; height: 100%;"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
        style="align-items: center; width: 500px;"
      >
        <span
          class="bp3-icon bp3-icon-error StyledProps--main StyledProps--color StyledProps--color-red500"
          icon="error"
        >
          <svg
            data-icon="error"
            height="32"
            viewBox="0 0 20 20"
            width="32"
          >
            <desc>
              error
            </desc>
            <path
              d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-align-center StyledProps--color StyledProps--color-red500"
        >
          error
        </p>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Retry
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FFNewSubscription loading 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main spinner"
  >
    <div
      class="StyledProps--font StyledProps--main PageSpinner--spinner StyledProps--flex StyledProps--flex-align-center-center"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main PageSpinner--content"
        style="align-items: center; width: 500px;"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
          data-icon="steps-spinner"
        >
          <svg
            height="32"
            viewBox="0 0 128 128"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M122.5 69.25H96.47a33.1 33.1 0 000-10.5h26.05a5.25 5.25 0 010 10.5z"
                fill="#ff8e02"
              />
              <path
                d="M112.04 97.83L89.47 84.8a33.1 33.1 0 005.25-9.1l22.57 13.03a5.25 5.25 0 01-5.28 9.1zM88.68 117.35L75.65 94.78a33.1 33.1 0 009.1-5.25l13.02 22.57a5.25 5.25 0 11-9.1 5.25zM58.7 122.57V96.5a33.1 33.1 0 0010.5 0v26.07a5.25 5.25 0 01-10.5 0zM30.1 112.1l13.04-22.57a33.1 33.1 0 009.1 5.25L39.2 117.35a5.25 5.25 0 11-9.1-5.25zM10.6 88.74L33.16 75.7a33.1 33.1 0 005.25 9.1L15.88 97.83a5.25 5.25 0 11-5.25-9.1z"
                fill="#ffddb3"
                fill-opacity="0.3"
              />
              <path
                d="M5.37 58.75h26.06a33.1 33.1 0 000 10.5H5.37a5.25 5.25 0 010-10.5z"
                fill="#ffd29a"
                fill-opacity="0.4"
              />
              <path
                d="M15.85 30.17L38.4 43.2a33.1 33.1 0 00-5.24 9.1L10.6 39.25a5.25 5.25 0 115.25-9.1z"
                fill="#ffc680"
                fill-opacity="0.5"
              />
              <path
                d="M39.2 10.65l13.03 22.57a33.1 33.1 0 00-9.1 5.25l-13-22.57a5.25 5.25 0 119.1-5.25z"
                fill="#ffbb67"
                fill-opacity="0.6"
              />
              <path
                d="M69.2 5.43V31.5a33.1 33.1 0 00-10.5 0V5.42a5.25 5.25 0 1110.5 0z"
                fill="#ffb04d"
                fill-opacity="0.7"
              />
              <path
                d="M97.77 15.9L84.75 38.47a33.1 33.1 0 00-9.1-5.25l13.03-22.57a5.25 5.25 0 119.1 5.25z"
                fill="#ffa535"
                fill-opacity="0.8"
              />
              <path
                d="M117.3 39.26L94.7 52.3a33.1 33.1 0 00-5.25-9.1l22.57-13.03a5.25 5.25 0 015.25 9.1z"
                fill="#ff991b"
                fill-opacity="0.9"
              />
              <animatetransform
                attributeName="transform"
                calcMode="discrete"
                dur="1200ms"
                repeatCount="indefinite"
                type="rotate"
                values="0 64 64;30 64 64;60 64 64;90 64 64;120 64 64;150 64 64;180 64 64;210 64 64;240 64 64;270 64 64;300 64 64;330 64 64"
              />
            </g>
          </svg>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-align-center StyledProps--color StyledProps--color-grey600"
        >
          Loading, please wait...
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`FFNewSubscription update licenses 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main"
  >
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-right-xlarge"
      style="width: 50%;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-weight-bold"
      >
        authSettings.costCalculator.developerLicenses
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
      >
        <div
          class="bp3-slider slider"
        >
          <div
            class="bp3-slider-track"
          >
            <div
              class="bp3-slider-progress"
              style="left: 0.00%; right: 100.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress bp3-intent-primary"
              style="left: 0.00%; right: 30.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress"
              style="left: 70.00%; right: 0.00%; top: 0px;"
            />
          </div>
          <div
            class="bp3-slider-axis"
          >
            <div
              class="bp3-slider-label"
              style="left: 0.00%;"
            >
              0
            </div>
            <div
              class="bp3-slider-label"
              style="left: 20.00%;"
            >
              10
            </div>
            <div
              class="bp3-slider-label"
              style="left: 40.00%;"
            >
              20
            </div>
            <div
              class="bp3-slider-label"
              style="left: 60.00%;"
            >
              30
            </div>
            <div
              class="bp3-slider-label"
              style="left: 80.00%;"
            >
              40
            </div>
            <div
              class="bp3-slider-label"
              style="left: 100.00%;"
            >
              50
            </div>
          </div>
          <span
            class="bp3-slider-handle"
            style="left: calc(70.00% - 0px);"
            tabindex="0"
          >
            <span
              class="bp3-slider-label"
            >
              35
            </span>
          </span>
        </div>
        <div
          class="TextInput--main"
        >
          <div
            class="bp3-input-group textInput"
          >
            <input
              class="bp3-input"
              type="text"
              value="35"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="Layout--vertical StyledProps--main StyledProps--padding-right-xlarge"
      style="width: 50%;"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline StyledProps--flex-justifyContent-start"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-weight-bold"
        >
          authSettings.costCalculator.maus
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-small"
        >
          , 
          100
          K
           
          authSettings.costCalculator.increments
        </p>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
      >
        <div
          class="bp3-slider slider"
        >
          <div
            class="bp3-slider-track"
          >
            <div
              class="bp3-slider-progress"
              style="left: 0.00%; right: 100.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress bp3-intent-primary"
              style="left: 0.00%; right: 0.00%; top: 0px;"
            />
            <div
              class="bp3-slider-progress"
              style="left: 100.00%; right: 0.00%; top: 0px;"
            />
          </div>
          <div
            class="bp3-slider-axis"
          >
            <div
              class="bp3-slider-label"
              style="left: 0.00%;"
            >
              0
            </div>
            <div
              class="bp3-slider-label"
              style="left: 20.00%;"
            >
              100
            </div>
            <div
              class="bp3-slider-label"
              style="left: 40.00%;"
            >
              200
            </div>
            <div
              class="bp3-slider-label"
              style="left: 60.00%;"
            >
              300
            </div>
            <div
              class="bp3-slider-label"
              style="left: 80.00%;"
            >
              400
            </div>
            <div
              class="bp3-slider-label"
              style="left: 100.00%;"
            >
              500
            </div>
          </div>
          <span
            class="bp3-slider-handle"
            style="left: calc(2000.00% - 0px);"
            tabindex="0"
          >
            <span
              class="bp3-slider-label"
            >
              10000
            </span>
          </span>
        </div>
        <div
          class="bp3-popover-wrapper DropDown--main dropdown"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class=""
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                data-testid="slider-dropdown"
                style="width: 70px;"
              >
                <div
                  class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                >
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                    data-testid="dropdown-value"
                    style="--text-line-clamp: 1;"
                  >
                      K
                  </p>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                >
                  <span
                    class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                    data-icon="main-chevron-down"
                  >
                    <svg
                      height="8"
                      viewBox="0 0 14 8"
                      width="8"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                        fill="#4d5969"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
